File: data/code_pairs/fortran/1668.f90
PROGRAM TestSCABS1
      COMPLEX Z
      REAL SCABS1

      ! Test cases
      Z = CMPLX(3.0, 4.0)
      PRINT *, 'SCABS1(3+4i) =', SCABS1(Z)

      Z = CMPLX(-5.0, -12.0)
      PRINT *, 'SCABS1(-5-12i) =', SCABS1(Z)

      END

      REAL FUNCTION SCABS1(Z)
      COMPLEX Z
      INTRINSIC ABS,AIMAG,REAL
      SCABS1 = ABS(REAL(Z)) + ABS(AIMAG(Z))
      RETURN
      END

File: data/code_pairs/cpp/1668.cpp
#include <iostream>
#include <complex>
#include <cmath>

float SCABS1(std::complex<float> z) {
    return std::abs(z.real()) + std::abs(z.imag());
}

int main() {
    // Test cases
    std::complex<float> z1(3.0, 4.0);
    std::cout << "SCABS1(3+4i) = " << SCABS1(z1) << std::endl;

    std::complex<float> z2(-5.0, -12.0);
    std::cout << "SCABS1(-5-12i) = " << SCABS1(z2) << std::endl;

    return 0;
}